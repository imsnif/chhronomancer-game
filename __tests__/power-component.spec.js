import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

jest.useFakeTimers()

test('Power progress indicator => renders correctly', () => {
  const Power = require('../components/power').default
  const tree = renderer.create(
    <Power name='foo' timeLeft='00:42:12' progress='50' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
