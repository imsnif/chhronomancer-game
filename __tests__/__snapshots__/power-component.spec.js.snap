// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Power progress indicator => renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#329932",
        "flex": 1,
        "fontFamily": "telegrama_raw",
        "fontSize": 6,
        "marginLeft": 9,
        "paddingTop": 2,
        "textAlign": "left",
      }
    }
  >
    foo
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#329932",
        "flex": 7,
        "fontFamily": "telegrama_raw",
        "fontSize": 6,
        "paddingTop": 2,
        "textAlign": "left",
      }
    }
  >
    | |==========================------------------------|  50% 00:42:12
  </Text>
</View>
`;
